neuralpp/symbolic/general_normalizer.py

```diff
@@ -52,18 +68,14 @@ def _normalize(expression: Expression, context: Z3SolverExpression) -> Expressio
                                       BasicQuantifierExpression(operation, index, constraint & ~condition, else_)),
                             context)
                     else:
-                        return if_then_else(condition,
-                                            _normalize(BasicQuantifierExpression(operation, index, constraint, then),
-                                                       context & condition),
-                                            _normalize(BasicQuantifierExpression(operation, index, constraint, else_),
-                                                       context & ~condition))
+                        return _normalize_conditional(condition,
```

Comment:
There's a bug in previous version: when `context & condition` or `context & ~condition` is not satisfiable, it should not be `_normalize()`d